# BCB/EEOB 546X, Computational Skills for Biological Data, Spring 2021 
### Wednesday/Friday, 1:10-2:30p.m., Virtual

## Course Website

[https://eeob-biodata.github.io/EEOB-BCB-546](https://eeob-biodata.github.io/EEOB-BCB-546)

The course website contains links to all of the information you will need to navigate this course. 

### Canvas Page

We will also provide some information via the course Canvas page (for content that cannot be shared publicly). 

[https://canvas.iastate.edu/courses/79905](https://canvas.iastate.edu/courses/79905)

## Instructors

All office hours are by appointment. If you need to contact any instructor, please send a direct message via [Slack](https://biodataskills.slack.com/).

* Dr. Matt Hufford
* Dr. Tracy Heath
* Dr. Dennis Lavrov

## Course Description

With the advent of big data across a number of sub-disciplines within biology, there is an increasing need for biologists to be comfortable and competent with command-line data processing and analysis. This course will develop basic skills necessary for biologists working with big data sets. Topics will include UNIX commands, scripting in R and Python, version control using Git and GitHub, use of high performance computing clusters, and writing effective data-management plans. These topics will be taught using a combination of lectures and computational exercises.

## Course Format

This course will be held virtually. A large proportion of our class meetings will be held synchronously via Zoom. These meetings will be recorded and the videos will be provided on Canvas. For some tutorials, we will provide instructional videos (on Canvas) and use our synchronous meetings to discuss any issues you encounter. 

You can find more details about our Zoom sessions here:

[https://canvas.iastate.edu/courses/79905/pages/zoom-sessions](https://canvas.iastate.edu/courses/79905/pages/zoom-sessions)

### Course Schedule

A detailed schedule the course can be found here:

[https://docs.google.com/spreadsheets/d/1tTXO1SLPC1Rva90x9e8FhzVk9-uBWvca9YQsC0Zf1V4/edit?usp=sharing](https://docs.google.com/spreadsheets/d/1tTXO1SLPC1Rva90x9e8FhzVk9-uBWvca9YQsC0Zf1V4/edit?usp=sharing)

## Textbook

Buffalo, V. ​Bioinformatics Data Skills​. Sebastopol, CA: O’Reilly Media, Inc., 2015. Print and eBook.

Iowa State University has purchased 6 seats for O’Reilly eBooks through Safari-ProQuest. T​his means that only 6 users at a time can access any/all of the O’Reilly books. Please feel free to utilize this resource, but be sure to close your browser window when done reading in order to free up a seat for other users. If you receive a message that the book is unavailable, keep checking back until a seat is available. A direct link to the book through the ISU library is found below:<br>
[http://proquest.safaribooksonline.com.proxy.lib.iastate.edu/9781449367480?uicode=iastate](http://proquest.safaribooksonline.com.proxy.lib.iastate.edu/9781449367480?uicode=iastate)

### Additional Text

We also recommend Allesina & Wilmes, which is available as an e-book through the ISU Libraries [[link to ebook](https://iowa-primo.hosted.exlibrisgroup.com/primo-explore/fulldisplay?docid=01IASU_ALMA21319744590002756&context=L&vid=01IASU&lang=en_US&search_scope=ComboPrimocentral&adaptor=Local%20Search%20Engine&tab=default_tab&query=any,contains,Computing%20skills%20for%20biologists&offset=0)]: 

Allesina and Wilmes. *Computing Skills for Biologists: A Toolbox*. Princeton University Press, 2019. 

## Correspondence

All correspondence should be conducted via Slack. You have been invited to join the BioDataSkills Slack team. We recommend that you download the ​[desktop application](https://slack.com/downloads/)​, which is available for Windows, Mac OS X, and Linux (there are also mobile apps). Via Slack, you can send group messages, private messages to any instructor (or all three at once) or class member, and hold discussions regarding lecture topics and assignments. It is also possible to upload files and bits of code on Slack. We have found this platform to be extremely useful for course facilitation and for collaborating on computational projects.


## Learning Outcomes​

By the end of this course, students should be able to:

1. Navigate through their computers, create and modify files and directories, and process
data using basic Unix commands.
2. Become familiar with basic R syntax and data structures and implement these in data
analysis and plotting.
3. Utilize the Python scripting language for more sophisticated data processing.
4. Become familiar with various genomic data types (range, sequence, and alignment data)
and learn how to write scripts and analysis pipelines for working with these data.
5. Become familiar with high performance computing resources at Iowa State as well as
how and when to employ these resources.
6. Explore additional resources/topics in computational biology including manuscript
preparation in LaTeX and Overleaf and creation of NSF-style Data Management Plans.

## Hardware and Software Requirements

In order to complete computational exercises and the group project for this course, students will need access to a Unix operating system. The Mac OS X operating system is based on Unix. Mac users can access a command line interface through the “Terminal” application (located in the `/Applications/Utilities` folder). For PC users, a Unix environment can be emulated using the “[​Git Bash](https://www.atlassian.com/git/tutorials/git-bash)​” program, by partitioning your hard drive and installing ​[Ubuntu](https://ubuntu.com/download)​, or by installing [Cygwin](https://www.cygwin.com/)​. These solutions for PC users may not be ideal for all course exercises and we can provide access to Mac computers if you would prefer to go this route. We will also demonstrate how to log into HPC-class, a computer cluster for coursework at ISU. When on HPC-class, you are in a Unix environment.

Links for PC Unix emulation:

* GitBash: [https://www.atlassian.com/git/tutorials/git-bash](https://www.atlassian.com/git/tutorials/git-bash)
* Ubuntu: [https://ubuntu.com/download](https://ubuntu.com/download)
* Cygwin: [https://www.cygwin.com](https://www.cygwin.com/)

Software requirements for various portions of the course will be communicated through the GitHub repository found at the following url:

[https://eeob-biodata.github.io/EEOB-BCB-546/software.html](https://eeob-biodata.github.io/EEOB-BCB-546/software.html)

## Grading

| Assignment   | Grade Percentage        |
|----------|----------|
| 1. Unix | 15% |
| 2. R | 15% |
| 3. Data Management Plan | 15% |
| 4. Python | 15% |
| 5. Group Project & Presentation | 40% |

Late work and plagiarism/other forms of academic misconduct will result in a zero grade for the relevant assignment. Students suspected of academic misconduct will be referred to the Dean of Students Office. For information on ISU’s Standards of Academic Integrity, please see the following website:

[https://www.studentconduct.dso.iastate.edu/academic/students](https://www.studentconduct.dso.iastate.edu/academic/students)

### Grading Scale

| Grade   | Percentage (p) |
|----------|----------|
| A | p ≥ 92 |
| A- | 90 ≤ p < 92 |
| B+ | 88 ≤ p < 90 |
| B | 82 ≤ p < 88 |
| B- | 80 ≤ p < 82 |
| C+ | 78 ≤ p < 80 |
| C | 72 ≤ p < 78 |
| C- | 70 ≤ p < 72 |
| D+ | 68 ≤ p < 70 |
| D | 62 ≤ p < 68 |
| D- | 60 ≤ p < 62 |
| F| p < 60 |


